crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -3229,6 +3300,85 @@ pub(crate) mod tests {
         );
     }
 
+    /// Check that register_operation correctly registers operations from MQTT and handles error conditions
+    #[tokio::test]
+    async fn test_register_entity() {
+        let tmp_dir = TempTedgeDir::new();
+        let mut config = c8y_converter_config(&tmp_dir);
+        config.enable_auto_register = false;
+
+        let (mut converter, _http_proxy) = create_c8y_converter_from_config(config);
+
+        // main device command
+        let main_device = converter.entity_store.main_device().clone();
+
+        let operation_topic = converter.mqtt_schema.topic_for(
+            &main_device,
+            &Channel::CommandMetadata {
+                operation: OperationType::Custom("my_operation".to_string()),
+            },
+        );
+        let operation_msg = MqttMessage::new(&operation_topic, "{}");
+
+        converter.convert(&operation_msg).await;
```

Comment:
With no assertion on the outcome of that call, the intention of the test isn't very clear. I was expecting the `114` message as the result of that new custom operation registration.

Reply:
You're right, and to be honest I wrote additional tests because I saw the test coverage decrease, so I meant to cover new branches I added, but in focusing on that I neglected to actually add assertions that verify the output is correct :facepalm:  I'll fix the test.

Reply:
fixed in 01ec205849